File: MMORTS-Client/Assets/Scripts/GameLogic/Unit.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace GameLogic
{
    public class Unit
    {
        public int uID { get; set; }
        public Player Owner { get; set; }
        public Vector3 UnitPosition { get; set; }
        public string UnitType { get; set; }
        public Vector3 Destination { get; set; }
    }
}

File: server/core/unit.py
"""
Module describing game units.
"""


class UnitType:
    def __init__(self, name, player, params):
        self.name = name
        self.player = player
        self.params = params


class Unit:
    def __init__(self, uid, player, unit_type, position):
        self.uid = uid
        self.type = unit_type
        self.player = player
        self.position = position
        self.destination = None

    def to_dict(self, private=None):
        result = {
            'uid': str(self.uid),
            'type': self.type.name,
            'position': self.position.to_list(),
        }
        if private == self.player:
            if self.destination:
                result.update({
                    'destination': self.destination.to_list()
                })
        return result
